lib/internal/url.js

```diff
@@ -847,17 +856,19 @@ class URLSearchParams {
             throw new errors.TypeError('ERR_INVALID_TUPLE', 'Each query pair',
                                        '[name, value]');
           }
-          const key = toUSVString(pair[0]);
-          const value = toUSVString(pair[1]);
-          this[searchParams].push(key, value);
+          this[searchParams].push(pair[0], pair[1]);
         }
       } else {
         // record<USVString, USVString>
+        // Need to use reflection APIs for full spec compliance.
         this[searchParams] = [];
-        for (var key of Object.keys(init)) {
-          key = toUSVString(key);
-          const value = toUSVString(init[key]);
-          this[searchParams].push(key, value);
+        for (const key of Reflect.ownKeys(init)) {
```

Comment:
@jasnell Did you mean `for-in`? I think that one doesn't give you the property symbols.